### PR TITLE
[pkg/ottl] Fix gocritic complaints

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Addresses gocritic complaints.

Related to #10466

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

File: pkg/ottl/contexts/ottldatapoints/datapoints.go [hunk 1]
```diff
@@ -434,16 +431,14 @@ func accessTimeUnixNano() ottl.StandardGetSetter[TransformContext] {
 func accessDoubleValue() ottl.StandardGetSetter[TransformContext] {
 	return ottl.StandardGetSetter[TransformContext]{
 		Getter: func(ctx TransformContext) interface{} {
-			switch ctx.GetDataPoint().(type) {
-			case pmetric.NumberDataPoint:
+			if _, ok := ctx.GetDataPoint().(pmetric.NumberDataPoint); ok {
```
Comment (evan-bradley) [inline pkg/ottl/contexts/ottldatapoints/datapoints.go:434]: Wouldn't it be better to assign the converted data point to a value so we don't have to convert it again on the following line?

Comment (TylerHelmuth) [reply pkg/ottl/contexts/ottldatapoints/datapoints.go:434]: Yes

File: pkg/ottl/contexts/ottldatapoints/datapoints.go [hunk 2]
```diff
@@ -434,17 +431,17 @@ func accessTimeUnixNano() ottl.StandardGetSetter[TransformContext] {
 func accessDoubleValue() ottl.StandardGetSetter[TransformContext] {
 	return ottl.StandardGetSetter[TransformContext]{
 		Getter: func(ctx TransformContext) interface{} {
-			switch ctx.GetDataPoint().(type) {
-			case pmetric.NumberDataPoint:
-				return ctx.GetDataPoint().(pmetric.NumberDataPoint).DoubleValue()
+			numberDataPoint, ok := ctx.GetDataPoint().(pmetric.NumberDataPoint)
```
Comment (evan-bradley) [inline pkg/ottl/contexts/ottldatapoints/datapoints.go:434]:
```suggestion
			if numberDataPoint, ok := ctx.GetDataPoint().(pmetric.NumberDataPoint); ok {
```

I think keeping the assignment in the `if` statement would be a bit more succinct.

**Review by evan-bradley — APPROVED** [review]

File: pkg/ottl/contexts/ottldatapoints/datapoints.go (hunk 2 above)
Comment (TylerHelmuth) [reply pkg/ottl/contexts/ottldatapoints/datapoints.go:434]: Youre right. I should stop multitasking

**Review by bogdandrutu — APPROVED** [review]

File: pkg/ottl/contexts/ottldatapoints/datapoints.go (hunk 2 above)
Comment (bogdandrutu) [reply pkg/ottl/contexts/ottldatapoints/datapoints.go]: I do think it is more succinct but harder to read. It is much easier to follow the code without having to go to the "branches".

**Review by codeboten — APPROVED** [review]

**Pull request merged**